skimage/draw/draw.py

```diff
@@ -520,7 +554,20 @@ def circle_perimeter(r, c, radius, method='bresenham', shape=None):
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
     """
-    return _circle_perimeter(r, c, radius, method, shape)
+    if not np.can_cast(r, np.integer) or not np.can_cast(c, np.integer):
+        raise TypeError("r and c must be integers")
```

Comment:
Removing these parameters from the Cython signature made this type checking necessary in order to keep the old behavior. Is this the best way to go about this?

Reply:
Why do r and c need to be integers?

Reply:
The old Cython signature on master declared these as `Py_ssize_t`. So technically we need to 
So raising a TypeError continues the old behavior. An alternative would be to cast/round them to integers but that would technically change the API.

Reply:
I think the `index` function will do what you want. It was introduced in python to define things that behave like Pyssize

Reply:
Thanks, using the `index` function now.